common/v2/features/AddAccount/AddAccountFlow.tsx

```diff
@@ -60,24 +62,45 @@ function AddAccountFlow() {
     goToNextStep();
   };
 
+  // Read the walletName parameter from the URL
+  const { match: { params: { walletName: walletNameFromURL } } } = props;
+
   const onWalletSelection = (name: WalletName) => {
+    // If wallet has been selected manually by user click, add the wallet name to the URL for consistency
+    if (name) {
+      props.history.replace(`/add-account/${name}`);
+    }
+
     setStoryName(name);
     // @ADD_ACCOUNT_TODO: This is equivalent to formDispatch call.
     // Maybe we can merge story and accountType to use only one?
     updateFormState({ type: ActionType.SELECT_ACCOUNT_TYPE, payload: { accountType: name } });
   };
 
-  const renderDefault = () => (
-    <ContentPanel className="" data-testid="Main-panel">
-      <div className="MainPanel">
-        <TransitionGroup>
-          <CSSTransition classNames="DecryptContent" timeout={500}>
-            <WalletList wallets={STORIES} onSelect={onWalletSelection} />
-          </CSSTransition>
-        </TransitionGroup>
-      </div>
-    </ContentPanel>
-  );
+  // If there is a valid walletName parameter in the URL, redirect to that wallet
+  if (walletNameFromURL) {
+    if (!walletNames.includes(walletNameFromURL)) {
+      props.history.replace('/add-account');
+    } else {
+      if (storyName !== walletNameFromURL) {
+        onWalletSelection(walletNameFromURL);
+      }
+    }
+  }
```

Comment:
FYI: in the future please avoid nested conditionals as much as possible.
Prefer:
```
if (walletNameFromURL 
   && walletNames.includes(walletNameFromURL)
   && storyName !== walletNameFromURL ) {
  onWalletSelection(walletNameFromURL)
} else if (walletNameFromURL 
  && !walletNames.includes(walletNameFromURL)) {
  props.history.replace('/add-account');
}
```